packages/api-contracts/ai-scan-api.yaml

```diff
@@ -237,17 +239,54 @@ components:
                     type: string
                 url:
                     type: string
-                error:
-                    type: string
-                code:
-                    type: string
                 scanResult:
                     $ref: '#/components/schemas/ScanResult'
-                run:
-                    $ref: '#/components/schemas/Run'
                 reports:
-                    $ref: '#/components/schemas/Reports'
-        ScanResultResponses:
-            type: array
-            items:
-                $ref: '#/components/schemas/ScanResultResponse'
+                    type: array
+                    items:
+                        $ref: '#/components/schemas/ScanReport'
+                run:
+                    $ref: '#/components/schemas/ScanRun'
+        WebApiError:
+            type: object
+            required:
+                - code
+                - message
+            properties:
+                code:
+                    type: string
+                    enum:
+                        - ResourceNotFound
+                        - InvalidResourceId
+                        - InvalidJsonDocument
+                        - RequestBodyTooLarge
+                        - InvalidURL
+                        - InternalError
+                        - MissingApiVersionQueryParameter
+                        - MissingContentTypeHeader
+                        - UnsupportedContentType
+                        - UnsupportedApiVersion
+                message:
+                    type: string
+        ScanRunError:
+            type: object
+            required:
+                - code
+                - message
+            properties:
+                code:
+                    type: string
+                    enum:
+                        - InternalError
+                message:
+                    type: string
+        ScanRunErrorResponse:
+            type: object
+            required:
+                - scanId
+                - error
+            properties:
+                scanId:
+                    type: string
```

Comment:
shouldnt this be just url? since a job is not created in this case?

Reply:
there are two cases:
1.) you submit a scan jb on invalid url. for this error response could be:
{url: string, error: WebApiError}
2.) you ask for the scan status on invalid guid. for this response could be:
{scanId: string, error: WebApiError}

Reply:
No, this is scan error, not web api error. Any scan run (failed/pass) will have scan id
1. Invalid URL will not have scan id. We reject it at the injection time. If URL is not accessible from a browser this is scan run error that is returned in run object of scan run result
2. Invalid guid is handled on HTTP request and will return web api error before any processing